service/k8sclient/k8client_test.go

```diff
@@ -36,75 +37,174 @@ func TestK8Client(t *testing.T) {
 	validKubeconfig, err := kubeCtl.Run(ctx, []string{"config", "view", "-o", "json"}, nil)
 	require.NoError(t, err)
 
-	client, err := NewK8Client(ctx, string(validKubeconfig))
+	client, err := New(ctx, string(validKubeconfig))
 	require.NoError(t, err)
 	t.Cleanup(func() {
 		err := client.Cleanup()
 		require.NoError(t, err)
 	})
 
-	name := "test-cluster"
-	_ = client.DeleteXtraDBCluster(ctx, name)
+	l := logger.Get(ctx)
 
-	for {
-		clusters, err := client.ListXtraDBClusters(ctx)
+	t.Run("XtraDB", func(t *testing.T) {
+		name := "test-cluster"
+		_ = client.DeleteXtraDBCluster(ctx, name)
+
+		for {
+			clusters, err := client.ListXtraDBClusters(ctx)
+			require.NoError(t, err)
+
+			l.Info(clusters)
+
+			if findXtraDBCluster(clusters, name) == nil {
+				break
+			}
+			time.Sleep(5 * time.Second)
+		}
+
+		err = client.CreateXtraDBCluster(ctx, &XtraDBParams{
+			Name: name,
+			Size: 2,
+		})
 		require.NoError(t, err)
+		for {
+			clusters, err := client.ListXtraDBClusters(ctx)
+			require.NoError(t, err)
 
-		if findCluster(clusters, name) == nil {
-			break
+			l.Info(clusters)
+
+			if cluster := findXtraDBCluster(clusters, name); cluster != nil && cluster.State == ClusterStateReady {
+				break
+			}
+			time.Sleep(5 * time.Second)
 		}
-		time.Sleep(5 * time.Second)
-	}
 
-	err = client.CreateXtraDBCluster(ctx, &XtraDBParams{
-		Name: name,
-		Size: 2,
-	})
-	require.NoError(t, err)
-	for {
+		err = client.UpdateXtraDBCluster(ctx, &XtraDBParams{
+			Name: name,
+			Size: 3,
+		})
+		require.NoError(t, err)
+		for {
+			clusters, err := client.ListXtraDBClusters(ctx)
+			require.NoError(t, err)
+
+			l.Info(clusters)
+
+			cluster := findXtraDBCluster(clusters, name)
+			if cluster != nil && cluster.State == ClusterStateReady {
+				assert.Equal(t, int32(3), cluster.Size)
+				break
+			}
+			time.Sleep(5 * time.Second)
+		}
+
+		err = client.DeleteXtraDBCluster(ctx, name)
+		require.NoError(t, err)
+		for {
+			clusters, err := client.ListXtraDBClusters(ctx)
+			require.NoError(t, err)
+
+			l.Info(clusters)
+
+			if findXtraDBCluster(clusters, name) == nil {
+				break
+			}
+			time.Sleep(5 * time.Second)
+		}
 		clusters, err := client.ListXtraDBClusters(ctx)
 		require.NoError(t, err)
+		assert.Nil(t, findXtraDBCluster(clusters, name))
+	})
 
-		if cluster := findCluster(clusters, name); cluster != nil && cluster.State == ClusterStateReady {
-			break
+	t.Run("PSMDB", func(t *testing.T) {
+		name := "test-cluster-psmdb"
+		_ = client.DeletePSMDBCluster(ctx, name)
+
+		for {
```

Comment:
This code is used multiple times to ensure the cluster has been deleted (or it doesn't exists initially).
Can we create a func and call it from all the places?

Reply:
The problem here that we use the same check only in two places and I don't think that extracting it make sense, but I'll take a look. Maybe we can figure out something.